x-pack/plugins/observability_solution/apm/server/utils/es_fields_mappings.ts

```diff
@@ -0,0 +1,814 @@
+    },
+    processor: {
+      event: normalizeValue<string>(fields[PROCESSOR_EVENT]),
+    },
+    agent: {
+      name: normalizeValue<AgentName>(fields[AGENT_NAME]),
+    },
+  };
+};
+
+// todo: pending #192337
+export const linkedParentsOfSpanMapping = (fields: Partial<Record<string, unknown[]>>) => {
+  if (!fields) return undefined;
+
+  return (fields[SPAN_LINKS_TRACE_ID] as string[]).map((v, index) => {
+    return {
+      trace: {
+        id: v,
+      },
+      span: {
+        id: fields[SPAN_LINKS_SPAN_ID]?.[index] ?? '',
+      },
+    };
+  }) as SpanLink[];
+};
+
+export const transactionMapping = (fields: Partial<Record<string, unknown[]>>) => {
+  if (!fields) return undefined;
+
+  return {
+    transaction: {
+      representative_count: normalizeValue<number>(fields[TRANSACTION_REPRESENTATIVE_COUNT]),
+      result: normalizeValue<string>(fields[TRANSACTION_RESULT]),
+      sampled: normalizeValue<boolean>(fields[TRANSACTION_SAMPLED]),
+      id: normalizeValue<string>(fields[TRANSACTION_ID]),
+      duration: {
+        us: normalizeValue<number>(fields[TRANSACTION_DURATION]),
+      },
+      type: normalizeValue<string>(fields[TRANSACTION_TYPE]),
+      name: normalizeValue<string>(fields[TRANSACTION_NAME]),
+    },
+    service: {
+      node: {
+        name: normalizeValue<string>(fields[SERVICE_NODE_NAME]),
+      },
+      language: {
+        name: normalizeValue<string>(fields[SERVICE_NODE_NAME]),
+      },
+      name: normalizeValue<string>(fields[SERVICE_NAME]),
+      framework: {
+        name: normalizeValue<string>(fields[SERVICE_FRAMEWORK_NAME]),
+      },
+    },
+    trace: {
+      id: normalizeValue<string>(fields[TRACE_ID]),
+    },
+    agent: {
+      name: normalizeValue<AgentName>(fields[AGENT_NAME]),
+      version: normalizeValue<string>(fields[AGENT_VERSION]),
+    },
+    event: {
+      success_count: normalizeValue<boolean>(fields[EVENT_SUCCESS_COUNT]),
+      outcome: normalizeValue<EventOutcome>(fields[EVENT_OUTCOME]),
+    },
+    processor: {
+      event: normalizeValue<'transaction'>(fields[PROCESSOR_EVENT]),
+      name: normalizeValue<'transaction'>(fields[PROCESSOR_NAME]),
+    },
+    data_stream: {
+      namespace: normalizeValue<string>(fields[DATA_STREAM_NAMESPACE]),
+      type: normalizeValue<string>(fields[DATA_STEAM_TYPE]),
+      dataset: normalizeValue<string>(fields[DATA_STREAM_DATASET]),
+    },
+    span: {
+      id: normalizeValue<string>(fields[SPAN_ID]),
+      links: linkedParentsOfSpanMapping(fields),
+    },
+    observer: {
+      hostname: normalizeValue<string>(fields[OBSERVER_HOSTNAME]),
+      type: normalizeValue<string>(fields[OBSERVER_TYPE]),
+      version: normalizeValue<string>(fields[OBSERVER_VERSION]),
+      version_major: normalizeValue<number>(fields[OBSERVER_VERSION_MAJOR]),
+    },
+    timestamp: {
+      us: normalizeValue<number>(fields[TIMESTAMP]),
+    },
+    '@timestamp': normalizeValue<string>(fields[AT_TIMESTAMP]),
+    labels: {
```

Comment:
do we need to map labels explicitly?

If so we'd need some generic logic here (actually labels are NOT nested objects and can be plainly mapped here as a key-value map)